google_chat_ros/README.md

```diff
@@ -0,0 +1,56 @@
+# The package for using Google chat services with ROS
+
+## What is this?
+Use Google Chat API with ROS.  
+![Screenshot from 2021-11-01 15-53-27](https://user-images.githubusercontent.com/27789460/139635911-66232c88-d3b9-4d7d-940e-966fbac9d800.png)  
+System components  
+![GoogleChatROS_system](https://user-images.githubusercontent.com/27789460/139635648-4ddbf9da-90e9-4b87-b958-ca996a8ffc4f.png)
+
+## How to use?
+### 1. Create a service account and private key
+See [Google Official Document](https://developers.google.com/chat/how-tos/service-accounts#step_1_create_service_account_and_private_key). Please ensure to get JSON credetial file and save it. DO NOT LOST IT!  
+For JSK members, all keys are available at [Google Drive](https://drive.google.com/drive/folders/1Enbbta5QuZ-hrUWdTjVDEjDJc3j7Abxo?usp=sharing). If you make new  API keys, please upload them here.
```

Comment:
Can you upload other robot's API to google drive?
I found only one spot API, but it seems that you use other robot's API.

Reply:
Yeah, it is okay to make by myself, but I want other people to make API keys to check whether the README is understandable, like @tkmtnt7000 . Can you try to make belka's one? []***

Reply:
OK. I'll try it.

Reply:
I suceeded in making an API key for belka and uploaded it.
README is understandable for me.

(As a side note, for JSK users, I think we need to be careful about the Google accounts that access projects of each robot. I  had a littele hard time accessing belka's project, so we should write accesible accounts somewhere like manuals.)

Reply:
@tkmtnt7000 Thank you for your advice, what you say is completely true. We have to wrap up all API documents

Reply:
I can send Google chat from both euslisp and rostopic!

The following is needed before executing ros command.
1. Activate Google chat API (Hangouts Chat API) with various settings. After activating, fill in the blank of infomation, check the all functions, check dialogflow button in connection setting, and choose authentication.
2. Add bot to chat rooms.
3. `sudo apt install python-googleapi`or`pip install google-python-api-client`

Reply:
@tkmtnt7000 
1 and 2 are maybe important, I'll add them ;)
Is 3 not solved with `rosdep install` ?

Reply:
@mqcmd196 
It seems that 3 is not solved with `rosdep install`  because `sudo pip2 -U install PyDrive` failed in my environment... This is an output `rosdep install`
<details>
<summary>rosdep install</summary>

```
tsukamoto@tsukamoto-p15s:~/google_chat_ws/src/jsk_3rdparty$ rosdep install -y -r --from-paths . --ignore-src
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
google_chat_ros: No definition of [python-googleapi] for OS version [bionic]
switchbot_ros: Cannot locate rosdep definition for [message_exectime]
Continuing to install resolvable dependencies...
executing command [sudo -H apt-get install -y python-oauth2client]
[sudo] tsukamoto のパスワード: 
パッケージリストを読み込んでいます... 完了
依存関係ツリーを作成しています                
状態情報を読み取っています... 完了
以下のパッケージが自動でインストールされましたが、もう必要とされていません:
  gir1.2-geocodeglib-1.0 libfwup1 libllvm8 libwayland-client0:i386 libwayland-server0:i386
  linux-hwe-5.4-headers-5.4.0-74 linux-hwe-5.4-headers-5.4.0-77 linux-hwe-5.4-headers-5.4.0-80
  linux-hwe-5.4-headers-5.4.0-81 linux-hwe-5.4-headers-5.4.0-84 linux-hwe-5.4-headers-5.4.0-86
  python3-pyaudio shim ubuntu-web-launchers
これを削除するには 'sudo apt autoremove' を利用してください。
以下の追加パッケージがインストールされます:
  python-fasteners python-monotonic python-rsa python-sqlalchemy python-sqlalchemy-ext
  python-uritemplate
提案パッケージ:
  python-sqlalchemy-doc python-psycopg2 python-pymysql python-fdb python-pymssql
以下のパッケージが新たにインストールされます:
  python-fasteners python-monotonic python-oauth2client python-rsa python-sqlalchemy
  python-sqlalchemy-ext python-uritemplate
アップグレード: 0 個、新規インストール: 7 個、削除: 0 個、保留: 0 個。
809 kB のアーカイブを取得する必要があります。
この操作後に追加で 4,826 kB のディスク容量が消費されます。
取得:1 http://jp.archive.ubuntu.com/ubuntu bionic/universe amd64 python-uritemplate all 0.6-1ubuntu1 [5,042 B]
取得:2 http://jp.archive.ubuntu.com/ubuntu bionic/main amd64 python-monotonic all 1.1-2 [5,262 B]
取得:3 http://jp.archive.ubuntu.com/ubuntu bionic/main amd64 python-fasteners all 0.12.0-3 [13.4 kB]
取得:4 http://jp.archive.ubuntu.com/ubuntu bionic/universe amd64 python-rsa all 3.4.2-1 [28.9 kB]
取得:5 http://jp.archive.ubuntu.com/ubuntu bionic/main amd64 python-sqlalchemy all 1.1.11+ds1-1ubuntu1 [682 kB]
取得:6 http://jp.archive.ubuntu.com/ubuntu bionic/universe amd64 python-oauth2client all 4.1.2-3 [61.4 kB]
取得:7 http://jp.archive.ubuntu.com/ubuntu bionic/main amd64 python-sqlalchemy-ext amd64 1.1.11+ds1-1ubuntu1 [13.4 kB]
809 kB を 0秒 で取得しました (9,515 kB/s)         
以前に未選択のパッケージ python-uritemplate を選択しています。
(データベースを読み込んでいます ... 現在 613553 個のファイルとディレクトリがインストールされています。)
.../0-python-uritemplate_0.6-1ubuntu1_all.deb を展開する準備をしています ...
python-uritemplate (0.6-1ubuntu1) を展開しています...
以前に未選択のパッケージ python-monotonic を選択しています。
.../1-python-monotonic_1.1-2_all.deb を展開する準備をしています ...
python-monotonic (1.1-2) を展開しています...
以前に未選択のパッケージ python-fasteners を選択しています。
.../2-python-fasteners_0.12.0-3_all.deb を展開する準備をしています ...
python-fasteners (0.12.0-3) を展開しています...
以前に未選択のパッケージ python-rsa を選択しています。
.../3-python-rsa_3.4.2-1_all.deb を展開する準備をしています ...
python-rsa (3.4.2-1) を展開しています...
以前に未選択のパッケージ python-sqlalchemy を選択しています。
.../4-python-sqlalchemy_1.1.11+ds1-1ubuntu1_all.deb を展開する準備をしています ...
python-sqlalchemy (1.1.11+ds1-1ubuntu1) を展開しています...
以前に未選択のパッケージ python-oauth2client を選択しています。
.../5-python-oauth2client_4.1.2-3_all.deb を展開する準備をしています ...
python-oauth2client (4.1.2-3) を展開しています...
以前に未選択のパッケージ python-sqlalchemy-ext を選択しています。
.../6-python-sqlalchemy-ext_1.1.11+ds1-1ubuntu1_amd64.deb を展開する準備をしています ...
python-sqlalchemy-ext (1.1.11+ds1-1ubuntu1) を展開しています...
python-sqlalchemy (1.1.11+ds1-1ubuntu1) を設定しています ...
python-rsa (3.4.2-1) を設定しています ...
python-monotonic (1.1-2) を設定しています ...
python-fasteners (0.12.0-3) を設定しています ...
python-sqlalchemy-ext (1.1.11+ds1-1ubuntu1) を設定しています ...
python-uritemplate (0.6-1ubuntu1) を設定しています ...
python-oauth2client (4.1.2-3) を設定しています ...
executing command [sudo -H pip2 install -U PyDrive]
Collecting PyDrive
  Using cached https://files.pythonhosted.org/packages/6b/2d/c8e052ba51099faee0bfe71d84f35bb1576e6910483cad46b840a122ca6c/PyDrive-1.3.1-py2-none-any.whl
Requirement already up-to-date: PyYAML>=3.0 in /usr/local/lib/python2.7/dist-packages (from PyDrive)
Collecting oauth2client>=4.0.0 (from PyDrive)
  Using cached https://files.pythonhosted.org/packages/95/a9/4f25a14d23f0786b64875b91784607c2277eff25d48f915e39ff0cff505a/oauth2client-4.1.3-py2.py3-none-any.whl
Collecting google-api-python-client>=1.2 (from PyDrive)
  Using cached https://files.pythonhosted.org/packages/83/fc/98045b8c5e0ba12929d423e9ff6b742951bb846707539b18f19b27c6ddc3/google_api_python_client-1.12.8-py2.py3-none-any.whl
Collecting httplib2>=0.9.1 (from oauth2client>=4.0.0->PyDrive)
  Downloading https://files.pythonhosted.org/packages/37/fd/525d4c267cffe4437405fdb883c57b895ec03746337dc5ca7489d40f60d0/httplib2-0.20.2.tar.gz (267kB)
    100% |████████████████████████████████| 276kB 6.7MB/s 
Collecting rsa>=3.1.4 (from oauth2client>=4.0.0->PyDrive)
  Using cached https://files.pythonhosted.org/packages/2d/d3/41b3db87f262debadb153900d4e6f8d61aa87187dd6fedd855ed24e8526d/rsa-4.7.1.tar.gz
rsa requires Python '>=3.5, <4' but the running Python is 2.7.17
ERROR: the following rosdeps failed to install
  pip: command [sudo -H pip2 install -U PyDrive] failed
  pip: Failed to detect successful installation of [PyDrive]

```
</details>

Reply:
Thanks @tkmtnt7000 ! Waiting for https://github.com/ros/rosdistro/pull/31124 being merged